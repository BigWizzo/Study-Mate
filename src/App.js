import { Route, Switch } from 'react-router-dom';
import SubjectList from './components/subjects/SubjectList';
import StudentLogin from './components/students/StudentLogin';
import Page404 from './components/Page404/Page404';
import ClockingsList from './components/clockings/ClockingsList';
import StudentCreate from './components/students/StudentCreate';
import StudentLogout from './components/students/StudentLogout';
import ClockingItemDetails from './components/clockings/ClockingItemDetails';
import SubjectNew from './components/subjects/SubjectNew';
import ClockingNew from './components/clockings/ClockingNew';
import ClockingEdit from './components/clockings/ClockingEdit';
import BottomNav from './components/navbar/BottomNav';
import Appheader from './components/header/Appheader';

function App() {
  return (
    <div className="">
      <Appheader />
      <div className="container-fluid">
        <Switch>
          <Route path="/" component={SubjectList} exact />
          <Route path="/login" component={StudentLogin} />
          <Route path="/signup" component={StudentCreate} />
          <Route path="/login" component={StudentLogout} />
          <Route path="/subjects/new" component={SubjectNew} />
          <Route path="/subjects/:id" component={ClockingsList} />
          <Route path="/clockings/new" component={ClockingNew} />
          <Route path="/clockings/:id" component={ClockingItemDetails} exact />
          <Route path="/clockings/:id/edit" component={ClockingEdit} exact />
          <Route component={Page404} />
        </Switch>
      </div>
      <BottomNav />
    </div>
  );
}

export default App;
