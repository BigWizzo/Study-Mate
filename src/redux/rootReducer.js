import { combineReducers } from 'redux';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import studentReducer from './students/studentReducer';
import subjectReducer from './subjects/subjectReducer';
import clockingReducer from './clockings/clockingReducer';

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['student', 'subjects', 'clockings'],
};

const rootReducer = combineReducers({
  student: studentReducer,
  subjects: subjectReducer,
  clockings: clockingReducer,
});

export default persistReducer(persistConfig, rootReducer);
