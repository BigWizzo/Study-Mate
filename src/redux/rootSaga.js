import { all, call } from 'redux-saga/effects';
import { studentSagas } from './students/studentSagas';
import { subjectSagas } from './subjects/subjectSagas';

export default function* rootSaga() {
  yield all([call(studentSagas), call(subjectSagas)]);
}
