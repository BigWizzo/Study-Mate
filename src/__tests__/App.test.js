import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { BrowserRouter } from 'react-router-dom';
import { store, persistor } from '../redux/store';
import App from '../App';

test('should render the Covid19-Catalogue-Board Application', () => {
  const component = renderer.create(
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </PersistGate>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
