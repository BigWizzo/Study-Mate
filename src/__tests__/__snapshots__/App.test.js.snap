// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Covid19-Catalogue-Board Application 1`] = `
Array [
  <div
    className="appheader text-center row g-0 align-items-center bg-blue"
  >
    <h3
      className="text-light"
    >
      Study Mate
    </h3>
  </div>,
  <div
    className="container-fluid"
  >
    <div
      className="container-fliuid"
    >
      <div
        className="row py-3 mb-2 bg-white"
      >
        <span
          className="col-2 text-center"
        >
          <i
            className="fas fa-angle-left"
          />
        </span>
        <span
          className="col-8 text-center"
        >
          21 September 2021
        </span>
        <span
          className="col-2 text-center"
        >
          <i
            className="fas fa-angle-right"
          />
        </span>
      </div>
      <h1
        className="text-center my-2"
      >
        Subjects
      </h1>
      <div
        className="row"
      >
        <div
          className="text-center"
        >
          Nothing to show yet. Please add some subjects
        </div>
      </div>
    </div>
  </div>,
]
`;
